src/main/java/net/dv8tion/jda/internal/audio/AudioPacket.java

```diff
@@ -227,24 +216,26 @@ public static AudioPacket decryptAudioPacket(AudioEncryption encryption, Datagra
                 return null;
         }
 
-        byte[] encodedAudio;
+        ByteBuffer encodedAudio = encryptedPacket.encodedAudio;
+        int length = encodedAudio.remaining();
+        int offset = encodedAudio.arrayOffset();
         switch (encryption)
         {
             case XSALSA20_POLY1305:
-                encodedAudio = encryptedPacket.getEncodedAudio();
+//                length = encodedAudio.remaining();
```

Comment:
We should uncomment this for clarity.
Not sure what to do with the `int length` declaration above though.

Reply:
I kept it as a comment for clarity, assigning it again is not really needed here. The declaration above is the full length and the special cases subtract the nonce length from it.